src/main/webapp/js/webrtc_adaptor.js

```diff
@@ -1292,31 +1299,53 @@ export class WebRTCAdaptor
 	}
 
 
-	turnOffLocalCamera() 
-	{
-		if (this.remotePeerConnection != null) {
-
-			var track = this.localStream.getVideoTracks()[0];
-			track.enabled = false;
-		}
-		else {
-			this.callbackError("NoActiveConnection");
-		}
-	}
-
-	turnOnLocalCamera() 
-	{
-		//If it started in playOnly mode and wants to turn on the camera
-		if(this.localStream == null){
-			this.navigatorUserMedia(this.mediaConstraints, stream =>{
-				this.gotStream(stream);
-			}, false);
-		}
-		else if (this.remotePeerConnection != null) {
-			var track = this.localStream.getVideoTracks()[0];
-			track.enabled = true;
-		}
-	}
+	turnOffLocalCamera(streamId) 
+	 {
+		//We need to send black frames within a time interval, because when the user turn off the camera,
+		//player can't connect to the sender since there is no data flowing. Sending a black frame in each 3 seconds resolves it.
+		this.blackFrameTimer = setInterval(() => {			
+			this.dummyCanvas.getContext('2d').fillRect(0, 0, 320, 240);
+		 	this.replacementStream = this.dummyCanvas.captureStream();
+		}, 3000);
+
+		 if (this.remotePeerConnection != null) {
+			 if(streamId != null || streamId != undefined){
+				this.updateVideoTrack(this.replacementStream, streamId, this.mediaConstraints, null, true);
+			 }
+			 else{
+				this.updateVideoTrack(this.replacementStream, this.publishStreamId, this.mediaConstraints, null, true);
```

Comment:
Make a single call to `updateVideoTrack` by the correct parameter. Calling the same method in two different places for the same purpose is not a good practice